en/products/codec.md

```diff
@@ -0,0 +1,384 @@
+# ActorCloud codec plugin instructions
+
+Codec plugin is used for encoding and decoding of messages between ActorCloud and devices
+
+## 1. Function definition
+
+Refer to [function definition](product.md#功能定义).
+
+## 2. Codec Plug-in Programming
+
+
+![codec](_assets/codec.png)
+
+### Writing scripts
+
+Scripts are written in `Python`, and the runtime environment is `Python 3.6`.
+
+Write script code with reference to the data format specification. After testing for correct encoding and decoding, copy the code into the input box of edit script.
+
+#### Analog input description
+
+- Topic: message topic
+- Message: JSON format, such as `{"temp":17}`，`[{"temp":17},{"hmd":80}]`
+
+ActorCloud can convert data types.
+
+### Running
+
+Click the `Running` button to check if the results are as expected.
+
+#### Running result description
+
+- Error: script error or validation error
+- Status_code: status code
+- Result: the result after processing which will be specified in the follows.

```

Comment:
Error -> error
Status_code -> status_code
Result -> result
You  can see the result below